### ISSUE TITLE
Markdown parser cutting words out

### ISSUE THREAD

**Issue by paladox** [vsch/flexmark-java#172]
Hi, trying to use flexmark-java over pegdown and im finding that it is cutting words out if you do the following

```pre
commitmessage.maxSubjectLength
:       Maximum length of the commit message's subject line.  Defaults
        to 50 if not specified or less than 0.
```

and this is what i see

```pre
commitmessage.maxSubjectLength
  aximum length of the commit message’s subject line. Defaults to 50 if not specified or less than 0.
```

Notice it cut of the M.

tryed converting it here https://gerrit-review.googlesource.com/#/c/gerrit/+/138774/

**Comment by vsch** [comment]
I passed the input through the [TitleExtract.java](https://github.com/vsch/flexmark-java/blob/master/flexmark-java-samples/src/com/vladsch/flexmark/samples/TitleExtract.java sample and I am not getting the truncation. Something else is at play in your code.

Here is what I am getting after adding the markdown to title extract sample input:

```markdown

# Plugin commit-message-length-validator

Vendor
:  Gerrit Code Review

Version
:  v2.15-rc2

commitmessage.maxSubjectLength
:       Maximum length of the commit message's subject line.  Defaults
        to 50 if not specified or less than 0.

## About

This plugin checks the length of a commit's commit message subject and body, and reports warnings and errors to the git client if the lentghts are exceeded.

## Documentation

* [Commit Message Length Configuration](config.md)
* More Items

```

```html
<h1><a href="#plugin-commit-message-length-validator" id="plugin-commit-message-length-validator">Plugin commit-message-length-validator</a></h1>
<dl>
  <dt>Vendor</dt>
  <dd>Gerrit Code Review</dd>
  <dt>Version</dt>
  <dd>v2.15-rc2</dd>
  <dt>commitmessage.maxSubjectLength</dt>
  <dd>Maximum length of the commit message&rsquo;s subject line.  Defaults to 50 if not specified or less than 0.</dd>
</dl>
<h2><a href="#about" id="about">About</a></h2>
<p>This plugin checks the length of a commit&rsquo;s commit message subject and body, and reports warnings and errors to the git client if the lentghts are exceeded.</p>
<h2><a href="#documentation" id="documentation">Documentation</a></h2>
<ul>
  <li><a href="config.md">Commit Message Length Configuration</a></li>
  <li>More Items</li>
</ul>
```

**Comment by paladox** [comment]
Doing something like

https://gerrit.googlesource.com/plugins/commit-message-length-validator/+show/master/src/main/resources/Documentation/config.md

That works in the old pegdown parser, but truncates it in the flexmark-java parser.

**Comment by vsch** [comment]
@paladox, I figured out the content and it is a bug in flexmark-java definitions extension when tabs are used for separating the marker from the text.

How do I get access to the raw file content, not rendered HTML. It will make it easier in the future to figure things out since tabs vs spaces and line endings can affect the parser.

**Comment by paladox** [comment]
ah thanks.

**Comment by vsch** [comment]
@paladox, it is always good to have the source that causes a problem because I can run a test against it and see if the cause is duplicated.

I fixed the bug in 0.28.2. Repo was updated. Maven may take some time to propagate the update.

**Comment by paladox** [comment]
Thankyou so much :)
